.github/workflows/publish-image.yml

```diff
@@ -72,26 +82,42 @@ jobs:
             -
                 name: Extract Docker metadata
                 id: meta
-                uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+                uses: docker/metadata-action@69f6fc9d46f2f8bf0d5491e4aabe0bb8c6a4678a
                 with:
                     images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
-
+            -
+                name: Set Version number
+                run: |
+                    VER=$(cat VERSION)
+                    echo "VERSION=$VER" >> $GITHUB_ENV
+            -
+                name: Set build datetime
+                run: |
+                    TIMESTAMP=$(date -u +'%Y-%m-%d T%H:%M:%SZ')
+                    echo "BUILD_DATE=$TIMESTAMP" >> $GITHUB_ENV
             # Build and push Docker image with Buildx
             # https://github.com/docker/build-push-action
             -
                 name: Build and push Docker image
                 id: build-and-push
-                uses: docker/build-push-action@v2
+                uses: docker/build-push-action@v3
                 with:
                     context: .
-                    push: true
-                    tags: ${{ env.IMAGE_NAME }}:${{ env.TAG }}
+                    platforms: linux/amd64,linux/arm64,linux/arm/v7
+                    push: ${{ github.event_name != 'pull_request' }}
+                    tags: |
+                      ${{ env.IMAGE_NAME }}:${{ env.TAG }}
+                      ${{ env.IMAGE_NAME }}:${{ env.VERSION }}
```

Comment:
Push to both the branch tag (using `latest` for the default branch) and the version tag from the contents of the version file

non default branches version file should be updated with non-standard semantic versions, e.g. 
- `dev-1.0.1`
- `indev-1.0.1`
- etc

As using normal versions will overwrite the default versions, for example.
A version 1.0.1 inside the indev branch will overwrite the main branch 1.0.1 on docker hub if allowed to push.